src/libs/execution_time_estimator/execution_time_estimator.cpp

```diff
@@ -96,4 +194,115 @@ Skill::parse_args(const std::string &args)
 	}
 }
 
+std::map<std::string, Skill>
+ExecutionTimeEstimator::get_skills_from_config(const std::string &path) const
+{
+	const int                                     ID       = 0;
+	const int                                     PROPERTY = 1;
+	std::unique_ptr<Configuration::ValueIterator> it(config_->search(path.c_str()));
+	std::map<std::string, std::string>            skill_strings;
+	while (it->next()) {
+		std::vector<std::string> skill_property =
+		  str_split(std::string(it->path()).substr(path.size()));
+		if (skill_property.size() != 2) {
+			break;
+		}
+		if (skill_property[PROPERTY] == "args") {
+			skill_strings[skill_property[ID]] += str_join(it->get_strings(), ',');
+		} else if (skill_property[PROPERTY] == "name") {
+			skill_strings[skill_property[ID]] =
+			  it->get_string() + "{" + skill_strings[skill_property[ID]];
```

Comment:
No this works, the `name` case prepends the name and `{`, the `args` case appends the args, hence the skill string will be built correctly. The only case that will cause a crash is if `args` is specified but `name` is not. It will cause: `13:50:21.736368 FawkesMainThread: [EXCEPTION] Unexpected skill string: 'shelf=.*}'
`, which i guess is fine.